mono/utils/mono-threads-api.h

```diff
@@ -133,6 +136,24 @@ For further explanation of what can and can't be done in GC unsafe mode:
 		MONO_STACKDATA (__gc_safe_unbalanced_dummy); \
 		mono_threads_enter_gc_safe_region_unbalanced_internal (&__gc_safe_unbalanced_dummy)
 
+#else
+
+#define MONO_ENTER_GC_UNSAFE do {
+
+#define MONO_EXIT_GC_UNSAFE ; } while (0)
```

Comment:
are you introducing a `;` where one is not needed? Same question for all 3 EXIT statements.

Reply:
They were needed or the code would not compile. Perhaps mono was using the macros incorrectly, but it seemed like constructs such as :
```
some_goto_label:
    MONO_EXIT_GC_UNSAFE;
```

required the semicolon.